Framework/API/inc/MantidAPI/DataProcessorAlgorithm.h

```diff
@@ -52,7 +52,7 @@ template <class Base> class MANTID_API_DLL GenericDataProcessorAlgorithm : publi
   /// MPI option. If false, we will use one job event if MPI is available
   bool m_useMPI;
   Workspace_sptr assemble(Workspace_sptr partialWS);
-  Workspace_sptr assemble(const std::string &partialWSName, const std::string &outputWSName);
+  Workspace_sptr assemble(const std::string &partialWSName, [[maybe_unused]] const std::string &outputWSName);
```

Comment:
I guess technically we don't need to specify the [[maybe_unused]] attribute in the header declaration as well, but we don't lose anything from having it there.